requirements.txt

```diff
@@ -22,7 +22,7 @@ orjson >= 3.7
 packaging >= 21.3
 pathspec >= 0.8.0
 pendulum >= 2.1.2
-pydantic >= 1.8.2, != 1.9.1
+pydantic >= 1.10.0
```

Comment:
This is required for JSON serdes to work with dataclasses. Otherwise, we get this failure

```
assert _Pydantic_MyDataclass_140486686115840(x=1, y='test') == MyDataclass(x=1, y='test')
```

If this Pydantic upgrade is an issue for our users, we can revert this and add custom handling for dataclasses _or_ throw an error when serializing dataclasses if the Pydantic version is not sufficient.